frameworks/hdfs/src/main/java/com/mesosphere/sdk/hdfs/scheduler/Main.java

```diff
@@ -102,6 +103,11 @@ private static ServiceSpec serviceSpecWithCustomizedPods(RawServiceSpec rawServi
 
         return DefaultServiceSpec.newBuilder(serviceSpec)
                 .pods(Arrays.asList(journal, name, zkfc, data))
+                .replacementFailurePolicy(
+                        ReplacementFailurePolicy.newBuilder()
+                                .permanentFailureTimoutMs(null)
```

Comment:
I understand why you have used a `null` here, but in general passing `null` around is bad. In this particular case, it will work out just fine. But, we should avoid doing this. For example, we can have `ReplacementFailurePolicy.disablePermanentFailureReplacement()` and `ReplacementFailurePolicy.enablePermanentFailureReplacement(). permanentFailureTimoutMs(1234)`.

Also, `permanentFailureTimoutMs` maps to `permanentFailureTimoutMins` internally, something that should also be addressed in a separate PR.